src/Microsoft.AspNet.Mvc.ModelBinding/ModelBindingContext.cs

```diff
@@ -119,6 +144,18 @@ public Type ModelType
         }
 
         /// <summary>
+        /// Gets or sets a model name which is explicitly set using an <see cref="IModelNameProvider"/>. 
+        /// <see cref="Model"/>.
+        /// </summary>
+        public string BinderModelName { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value which represents the <see cref="BindingSource"/> associated with the 
+        /// <see cref="Model"/>.
+        /// </summary>
+        public BindingSource BindingSource { get; set; }
+
+        /// <summary>
```

Comment:
I had a hunch that might be the case. :+1: on locking this set of design changes down first